crates/router/src/connector/stripe/transformers.rs

```diff
@@ -1450,28 +1450,34 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for PaymentIntentRequest {
 
 fn get_payment_method_type_for_saved_payment_method_payment(
     item: &types::PaymentsAuthorizeRouterData,
-) -> Result<StripePaymentMethodType, error_stack::Report<errors::ConnectorError>> {
-    let stripe_payment_method_type = match item.recurring_mandate_payment_data.clone() {
-        Some(recurring_payment_method_data) => {
-            match recurring_payment_method_data.payment_method_type {
-                Some(payment_method_type) => StripePaymentMethodType::try_from(payment_method_type),
-                None => Err(errors::ConnectorError::MissingRequiredField {
-                    field_name: "payment_method_type",
+) -> Result<Option<StripePaymentMethodType>, error_stack::Report<errors::ConnectorError>> {
+    if item.payment_method == api_enums::PaymentMethod::Card {
+        Ok(None) //stripe takes ["Card"] as default
```

Comment:
better to be explicit rather than depend on defaults, pass payment method type as card here